tests/ChatTests/Forward.hs

```diff
@@ -0,0 +1,501 @@
+testForwardContactToContact =
+  testChat3 aliceProfile bobProfile cathProfile $
+    \alice bob cath -> do
+      connectUsers alice bob
+      connectUsers alice cath
+      connectUsers bob cath
+
+      alice #> "@bob hi"
+      bob <# "alice> hi"
+      msgId <- lastItemId alice
+      bob #> "@alice hey"
+      alice <# "bob> hey"
+
+      alice ##> ("/_forward @2 @3 " <> msgId)
+      alice <# "@cath -> from you"
+      alice <## "      hi"
+      cath <# "alice> -> forwarded"
+      cath <## "      hi"
+
+      alice `send` "> @bob -> @cath hey"
+      alice <# "@cath -> from bob (contact id: 2)"
+      alice <## "      hey"
+      cath <# "alice> -> forwarded"
+      cath <## "      hey"
+
+      -- read chat
+      alice ##> "/tail @cath 2"
+      alice <# "@cath -> from you"
+      alice <## "      hi"
+      alice <# "@cath -> from bob (contact id: 2)"
+      alice <## "      hey"
+
+      cath ##> "/tail @alice 2"
+      cath <# "alice> -> forwarded"
+      cath <## "      hi"
+      cath <# "alice> -> forwarded"
+      cath <## "      hey"
+
+      -- item info
+      alice ##> "/item info @cath hey"
+      alice <##. "sent at: "
+      alice <## "message history:"
+      alice .<## ": hey"
+      alice <##. "forwarded from chat item id:"
+
+testForwardContactToGroup :: HasCallStack => FilePath -> IO ()
+testForwardContactToGroup =
+  testChat3 aliceProfile bobProfile cathProfile $
+    \alice bob cath -> do
+      connectUsers alice bob
+      createGroup2 "team" alice cath
+
+      alice #> "@bob hi"
+      bob <# "alice> hi"
+      bob #> "@alice hey"
+      alice <# "bob> hey"
+
+      alice `send` ">> @bob -> #team hi"
+      alice <# "#team -> from you"
+      alice <## "      hi"
+      cath <# "#team alice> -> forwarded"
+      cath <## "      hi"
+
+      alice `send` "> @bob -> #team hey"
+      alice <# "#team -> from bob (contact id: 2)"
+      alice <## "      hey"
+      cath <# "#team alice> -> forwarded"
+      cath <## "      hey"
+
+testForwardContactToNotes :: HasCallStack => FilePath -> IO ()
+testForwardContactToNotes =
+  testChat2 aliceProfile bobProfile $
+    \alice bob -> do
+      createCCNoteFolder alice
+      connectUsers alice bob
+
+      alice #> "@bob hi"
+      bob <# "alice> hi"
+      bob #> "@alice hey"
+      alice <# "bob> hey"
+
+      alice `send` ">> @bob -> * hi"
+      alice <# "* -> from you"
+      alice <## "      hi"
+
+      alice `send` "> @bob -> * hey"
+      alice <# "* -> from bob (contact id: 2)"
+      alice <## "      hey"
+
+testForwardGroupToContact :: HasCallStack => FilePath -> IO ()
+testForwardGroupToContact =
+  testChat3 aliceProfile bobProfile cathProfile $
+    \alice bob cath -> do
+      createGroup2 "team" alice bob
+      connectUsers alice cath
+
+      alice #> "#team hi"
+      bob <# "#team alice> hi"
+      bob #> "#team hey"
+      alice <# "#team bob> hey"
+
+      alice `send` "> #team -> @cath hi"
+      alice <# "@cath -> from you"
```

Comment:
how will it show with /tail?

maybe

@cath -> you from #team

for user's messages

or

@cath -> from #team

for other messages

Reply:
currently it's like this:

```
alice <# "@cath -> from you"
alice <## "      hi"
alice <# "@cath -> from bob (contact id: 2)"
alice <## "      hey"
```

id - because name is not a unique identifier in cli

Reply:
with latest commit changed to:

```
alice <# "@cath <- from you"
alice <## "      hi"
alice <# "@cath <- from bob (contact id: 2)"
alice <## "      hey"
```